bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/api/ShadePosition.java

```diff
@@ -190,56 +154,196 @@ private State getPosition1(CoordinateSystem coordSys) {
+     *
+     * @return true if the ShadePosition supports tilt anywhere.
+     */
+    public boolean jsonTiltAnywhere() {
+        return ((PRIMARY_ZERO_IS_CLOSED.equals(posKind1)) && (VANE_TILT_COORDS.equals(posKind2))
+                || ((PRIMARY_ZERO_IS_CLOSED.equals(posKind2) && (VANE_TILT_COORDS.equals(posKind1)))));
+    }
+
+    /**
+     * Set the shade's position for the given actuator class resp. coordinate system.
+     *
+     * @param shadeCapabilities the shade Thing capabilities.
+     * @param posKindCoords the actuator class (coordinate system) whose state is to be changed.
+     * @param percent the new position value.
+     * @return this object.
+     */
+    public ShadePosition setPosition(Capabilities shadeCapabilities, CoordinateSystem posKindCoords, int percent) {
+        // if necessary swap the order of position1 and position2
+        if (PRIMARY_ZERO_IS_CLOSED.equals(posKind2) && !PRIMARY_ZERO_IS_CLOSED.equals(posKind1)) {
+            final Integer posKind2Temp = posKind2;
+            final Integer position2Temp = position2;
+            posKind2 = Integer.valueOf(posKind1);
+            position2 = Integer.valueOf(position1);
+            posKind1 = posKind2Temp != null ? posKind2Temp.intValue() : NONE.ordinal();
+            position1 = position2Temp != null ? position2Temp.intValue() : 0;
+        }
+
+        // delete position2 if it has an invalid position kind
+        if (ERROR_UNKNOWN.equals(posKind2) || NONE.equals(posKind2)) {
+            posKind2 = null;
+            position2 = null;
+        }
+
+        // logic to set either position1 or position2
+        switch (posKindCoords) {
+            case PRIMARY_ZERO_IS_CLOSED:
+                if (shadeCapabilities.supportsPrimary()) {
+                    setPosition1(shadeCapabilities, posKindCoords, percent);
+                }
+                break;
+
+            case SECONDARY_ZERO_IS_OPEN:
+                if (shadeCapabilities.supportsSecondary()) {
+                    if (shadeCapabilities.supportsPrimary()) {
+                        setPosition2(shadeCapabilities, posKindCoords, percent);
+                    } else {
+                        setPosition1(shadeCapabilities, posKindCoords, percent);
+                    }
+                }
+                break;
+
+            case VANE_TILT_COORDS:
+                if (shadeCapabilities.supportsPrimary()) {
+                    if (shadeCapabilities.supportsTiltOnClosed()) {
+                        setPosition1(shadeCapabilities, posKindCoords, percent);
+                    } else if (shadeCapabilities.supportsTiltAnywhere()) {
+                        setPosition2(shadeCapabilities, posKindCoords, percent);
+                    }
+                } else if (shadeCapabilities.supportsTiltAnywhere()) {
+                    setPosition1(shadeCapabilities, posKindCoords, percent);
+                }
+                break;
+
             default:
```

Comment:
Remove empty `default`.

Reply:
The `switch()` statement runs over an enum which includes values `NONE` and `ERROR` that shall not be handled; and I simply put in the empty `default` to get rid of the compiler warnings; the actual code runs no differently with or without it.

Reply:
Sorry, I missed that, just looked strange. Would it be an idea to explicitly provide cases for NONE and ERROR with a comment that they are not handled on purpose? In that case the compiler would provide a warning if a new enum value would be added without adapting this code accordingly.

Reply:
> explicitly provide cases for NONE and ERROR

Good idea. Will do.